samples/authUserFlow.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(projectId, clientId, clientSecret, refreshToken) {
+  // [START bigquery_auth_user_flow]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'PROJECT ID'
+  // const clientId = 'CLIENT ID'
+  // const clientSecret = 'CLIENT SECRET'
+  // const refreshToken = 'REFRESH TOKEN'
+
+  const credentials = {
+    type: 'authorized_user',
+    client_id: clientId,
+    client_secret: clientSecret,
+    refresh_token: refreshToken,
+  };
+  // [END bigquery_auth_user_flow]
+  // [START bigquery_auth_user_query]
+  async function query() {
+    const {BigQuery} = require('@google-cloud/bigquery');
+
+    const bigquery = new BigQuery({
+      projectId,
+      credentials,
+    });
+
+    // Queries the U.S. given names dataset for the state of Texas.
+    const query = `SELECT name
+            FROM \`bigquery-public-data.usa_names.usa_1910_2013\`
+            WHERE state = 'TX'
+            LIMIT 100`;
+
+    // For all options, see https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query
+    const options = {
+      query: query,
+      // Location must match that of the dataset(s) referenced in the query.
+      location: 'US',
```

Comment:
We're removing all the location options from our query samples in Python because the location auto-detect works in every region now for queries.